libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteProgressEx.kt

```diff
@@ -45,8 +46,14 @@ suspend fun RouteProgress.hasUnexpectedUpcomingClosures(): Boolean =
             }
 
             val silentWaypoints = routeLeg.silentWaypoints()
+            val expectedLegClosures = navigationRoute.expectedClosures
+                .getOrNull(routeLegIndex).orEmpty()
 
             routeLeg.closures()?.forEach { closure ->
+                if (closure in expectedLegClosures) {
+                    // skipping expected closures
+                    return@forEach
+                }
```

Comment:
I think this code will handle this case well. But maybe it makes sense to move those closures to the place where we check closures on the current leg and call them `unavoidableClosures` so that it's easier to understand what is the purpose of this logic. It was hard for me understand the case when closure can't be avoided as there is no legal way around it.

Reply:
But that's exactly the place where we check closures on the current leg. Which place do you mean?
"Unavoidable" makes sense, I'll rename.